builder/hcloud/config.go

```diff
@@ -47,6 +47,7 @@ type Config struct {
 	SSHKeys        []string          `mapstructure:"ssh_keys"`
 	SSHKeysLabels  map[string]string `mapstructure:"ssh_keys_labels"`
 	Networks       []int64           `mapstructure:"networks"`
+	PublicIPv4     string            `mapstructure:"public_ipv4_address"`
```

Comment:
@apricote I think we should use the ID or name of the primary IP instead of using the address, to keep it consistent with the rest of the config. What do you think?

Reply:
I would actually prefer that as well. How do you suggest to determine if we have an ID or an IP address to choose which hcloud API to use to get the Primary IP?

Reply:
We could use [`netip.ParseAddr()`](https://pkg.go.dev/net/netip#ParseAddr) to check if its a valid IP and fallback to `client.PrimaryIP.Get()` if it could not be parsed as an IP.

Reply:
The simple option I guess is to try `strconv.Atoi(c.PublicIPv4)` and branch on error

Reply:
That already happens in `client.PrimaryIP.Get()` for ID & Name: https://github.com/hetznercloud/hcloud-go/blob/0821c078900910fa9e3ca6c6c0af48a73f00c7c6/hcloud/primary_ip.go#L208-L215

Reply:
I've also added tests for the various failure scenarios now.

Reply:
strconv.Atoi prevents us from using the name (ID or name) of the primary IP, I'd rather go with the netip.ParseAddr.

To be feature complete, the same must probably be implemented for IPV6 as well